microceph/ceph/keyring.go

```diff
@@ -106,3 +106,39 @@ func parseKeyring(path string) (string, error) {
 
 	return cephSecret, nil
 }
+
+// CreateKey creates a client key and returns the said key hash without saving it as a file.
+func CreateKey(clientName string, caps ...[]string) (string, error) {
+	args := []string{
+		"auth",
+		"get-or-create",
+		fmt.Sprintf("client.%s", clientName),
```

Comment:
I would suggest sanitizing the client name beforehand: If the client name already starts with `client.`, then use it as is, otherwise construct it as you do. I recommend this because it's very easy to trip over this interface - it has happened to me in the past.

Reply:
The client names come from the remote database (remotes are the clients in this case).